Fix HumanName getNameAsSingleString never returning text element

amir-kt opened the pull request:
fixing https://github.com/hapifhir/org.hl7.fhir.core/issues/1206

dotasek reviewed (changes requested):
This fix looks fine. 

Please review the additional changes in [this branch](https://github.com/hapifhir/org.hl7.fhir.core/tree/amir-kt/master), confirm that they are correct, and merge them into this PR.

amir-kt commented:
merged [your branch](https://github.com/hapifhir/org.hl7.fhir.core/tree/amir-kt/master) with this PR. I removed a line in the unit test that defined an unused variable. 

I also fixed the same issue in r5 and added your unit tests to it too. I'm not sure what the other environments are or whether you'd like to implement the fix there too.

dotasek commented:
Thank you @amir-kt for merging that test. It was also a very good idea to check r5; that's something I should have noticed. I will look at the other FHIR versions and update them accordingly.

dotasek reviewed (changes requested):
@amir-kt I checked the previous versions and did the updates where applicable:

https://github.com/hapifhir/org.hl7.fhir.core/tree/amir-kt/master

Could you review these changes, merge them into this request, and then we can merge this into master and close.

dstu2016may passed without any alterations because that field is automatically instantiated in getFamily() in that version.

dstu2 has no getNameAsSingleString method.

amir-kt commented:
Awesome, reviewed and merged. Thanks @dotasek!

dotasek commented:
@amir-kt I'm going to merge this PR once we repair our pipeline issues, but in reviewing it with @grahamegrieve, he pointed out that this construction is not necessarily ideal.

We will follow up by modifying the logic as described in this issue: https://github.com/hapifhir/org.hl7.fhir.core/issues/1214

Please take a look at that issue and comment, as it will change the behaviour we defined in this PR.

dotasek commented:
I've run this without the tests that are failing on Master on a Windows machine. I'm going ahead with the merge.
